utils/clean_txt.py

```diff
@@ -2,59 +2,108 @@
 """A command line utility to clean text files generated from PDFs for Praxis-NLP."""
 import argparse
 from pathlib import Path
-from sys import stderr
 import re
 
-BRACKET_REGEX = re.compile(r"\s?\[\w*\]\s?|\s?\(\w*\)\s?")
-BIB_REGEX = re.compile(r"(Bibliography)|(Works Cited)|(Source Extracts)", re.IGNORECASE)
+# match expressions in square or round brackets containing alphanumeric
+# characters or spaces
+BRACKET_REGEX = re.compile(r'\s?\[(\w|\s|,)*\]|\s?\((\w|\s|,)*\)')
+BIB_REGEX = re.compile(
+    r'(Bibliography)|(Works Cited)|(Source Extracts)|(Appendix(:|\n))', re.IGNORECASE
+)
 EXTRA_WHITESPACE = re.compile(r" [a-z\)\],;]+\n")
 
 
+def remove_non_ascii(s):
+    """Re-encodes s as ASCII"""
+    s.encode('ascii', 'ignore')
+    return s
+
+
 def remove_citations(s, verbose=False):
-    """Removes anything between brackets, parentheses, or after a works cited heading"""
-    lines=[]
+    """Removes anything between brackets, or after a bibliograhpy heading"""
+    lines = []
     characters_written = 0
     characters_read = 0
 
     for line_number, line in enumerate(s.split('\n')):
 
         if re.search(BIB_REGEX, line):
             if verbose:
-                print(f"Truncated at line {line_number}: {line}", file=stderr)
+                print(f'Truncated at line {line_number}: {line}')
             break
-        (new_line, __) = re.subn(BRACKET_REGEX, "", line)
+        (new_line, __) = re.subn(BRACKET_REGEX, '', line)
         lines.append(new_line)
         characters_read += len(line)
         characters_written += len(new_line)
 
-    characters_read += sum(len(line) for line in s)
-    return ("\n".join(lines), (characters_read-characters_written)/characters_read)
+    return ('\n'.join(lines), (characters_read - characters_written) / characters_read)
+
 
 def fix_whitespace(s):
     """Removes superfluous lines and newlines"""
-    s = "\n".join(line.strip(" -\t\n") for line in s.split("\n") if line.strip(" -\t\n123456789."))
-    s, __ = re.subn(EXTRA_WHITESPACE, " ", s)
-    return s.replace("\u2000", "")
+    s = '\n'.join(
+        line.strip(' -\t\n') for line in s.split('\n') if line.strip(' -\t\n123456789.')
+    )
+    s, __ = re.subn(EXTRA_WHITESPACE, ' ', s)
+
+    # remove weird characters
+    s = s.replace(u'\u200b', '')
```

Comment:
Redundant by remove_non_ascii, it will already be stripped out.

Reply:
True, although I didn't realize because the ascii encoding is broken...